src/request/component-models/ComponentModel.ts

```diff
@@ -0,0 +1,7 @@
+'use strict';
+
+import { IComponentModel } from './IComponentModel';
+
+export abstract class ComponentModel implements IComponentModel {
+    public abstract init(request);
+}
```

Comment:
Is it still worth having this instead of just implementing the interface directly.